### PR TITLE
fix: marketo bulk upload edits

### PR DESCRIPTION
**Pull request by shrouti1507**
# Description

resolves INT-157
Fresh PR raised on the basis of https://github.com/rudderlabs/rudder-server/pull/3719

## Linear Ticket

[< Linear Link >](https://linear.app/rudderstack/issue/INT-157/marketo-bulk-upload-sanity-and-find-bugs)

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go [hunk 1]
```diff
@@ -111,7 +112,16 @@ func (b *MarketoBulkUploader) generateFailedPayload(jobs []*jobsdb.JobT, importI
 }
 
 func (b *MarketoBulkUploader) GetUploadStats(UploadStatsInput common.GetUploadStatsInput) common.GetUploadStatsResponse {
-	failedJobUrl := UploadStatsInput.FailedJobURLs
+	var finalURL string
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go]: done

File: router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go [hunk 2]
```diff
@@ -147,76 +145,30 @@ func (b *MarketoBulkUploader) GetUploadStats(UploadStatsInput common.GetUploadSt
...
-	if errFailed != nil || errWarning != nil || errSuccess != nil {
-		b.logger.Errorf("[Batch Router] Failed to get job IDs for Dest Type %v with metata %v", "MARKETO_BULK_UPLOAD", metadata)
-		return common.GetUploadStatsResponse{
-			StatusCode: 500,
-		}
-	}
 
-	failedReasons, gotFailedReason := metadata["failedReasons"].(map[string]interface{})
-	warningReasons, gotWarningReason := metadata["warningReasons"].(map[string]interface{})
+	failedReasons := failedJobsResponse.Metadata.FailedReasons
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go:163]: As verified over call, this is a structure and metadata should be always present.

File: router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go [hunk 3]
```diff
@@ -310,7 +325,7 @@ func (b *MarketoBulkUploader) Upload(asyncDestStruct *common.AsyncDestinationStr
 				DestinationID: destinationID,
 			}
 		}
-		successfulJobIDs, failedJobIDsTrans := extractJobStats(responseStruct.Metadata, importingJobIDs)
+		successfulJobIDs, failedJobIDsTrans := extractJobStats(responseStruct.Metadata, importingJobIDs, "200")
```
Comment (lvrach) [inline router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go:328]: [nit-pick]
```suggestion
		successfulJobIDs, failedJobIDsTrans := extractJobStats(responseStruct.Metadata, importingJobIDs, http.StatusOK)
```

File: router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go [hunk 4]
```diff
@@ -336,17 +351,22 @@ func (b *MarketoBulkUploader) Upload(asyncDestStruct *common.AsyncDestinationStr
 			"module":   "batch_router",
 			"destType": destType,
 		})
-		abortedJobIDs, failedJobIDsTrans := extractJobStats(responseStruct.Metadata, importingJobIDs)
+		abortedJobIDs, failedJobIDsTrans := extractJobStats(responseStruct.Metadata, importingJobIDs, "400")
```
Comment (lvrach) [inline router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go:354]: [nit-pick]


```suggestion
		abortedJobIDs, failedJobIDsTrans := extractJobStats(responseStruct.Metadata, importingJobIDs, http.StatusBadRequest)
```

**Review by lvrach — APPROVED** [review]

**Review by fracasula — APPROVED** [review]

**Review by chandumlg — APPROVED** [review]

**Pull request merged**